src/blocks/nordvpn.rs

```diff
@@ -0,0 +1,219 @@
+        .await?;
+
+    let mut widget = Widget::new();
+
+    let format_connected = config.format_connected.with_default(" VPN: $icon ")?;
+    let format_disconnected = config.format_disconnected.with_default(" VPN: $icon ")?;
+
+    loop {
+        let status = get_current_network_status().await?;
+
+        widget.set_values(map!(
+            "icon" => Value::icon(api.get_icon(status.icon())?),
+            "country" => match &status {
+               Status::Connected(country)  => Value::text(country.clone()),
+               _ => Value::text(String::default()),
+            }
+        ));
+
+        let state_connected = config.state_connected.as_deref().unwrap_or("info");
+        let state_disconnected = config.state_disconnected.as_deref().unwrap_or("idle");
+
+        widget.state = match status {
+            Status::Connected(_) => {
+                widget.set_format(format_connected.clone());
+                State::from_str(state_connected).unwrap()
+            }
+            Status::Disconnected => {
+                widget.set_format(format_disconnected.clone());
+                State::from_str(state_disconnected).unwrap()
+            }
+            Status::Error => {
+                widget.set_format(format_disconnected.clone());
+                State::Critical
+            }
+        };
+
+        api.set_widget(&widget).await?;
+
+        select! {
+            _ = sleep(config.interval.0) => (),
+            event = api.event() => {
+                match event {
+                    Action(a) if a == "toggle" => {
+                        match status {
+                            Status::Connected(_) => run_network_command("disconnect").await?,
+                            Status::Disconnected => run_network_command("connect").await?,
+                            Status::Error => (),
+                        }
+                    }
+                    _ => (),
+                }
+            }
+        }
+    }
+}
+
+async fn get_current_network_status() -> Result<Status> {
+    let stdout = Command::new("nordvpn")
+        .args(["status"])
+        .output()
+        .await
+        .error("Problem running nordvpn command")?
+        .stdout;
+
+    let stdout = String::from_utf8(stdout).error("nordvpn produced non-UTF8 output")?;
+    let line_status = find_line(&stdout, "Status:").await;
+    let line_country = find_line(&stdout, "Country:").await;
+    if line_status.is_none() {
+        return Ok(Status::Error);
+    }
+    let line_status = line_status.unwrap();
+
+    if line_status.ends_with("Disconnected") {
+        return Ok(Status::Disconnected);
+    } else if line_status.ends_with("Connected") {
+        match line_country {
+            Some(country_line) => {
+                let country = country_line
+                    .split(": ")
+                    .collect::<Vec<&str>>()
+                    .last()
+                    .unwrap()
+                    .to_string();
```

Comment:
There's [`rsplit`](https://doc.rust-lang.org/stable/std/primitive.str.html#method.rsplit) method on `str` that does this in fewer steps.

Reply:
See [here](https://github.com/greshake/i3status-rust/pull/1725/commits/2e502dcd794ea28a7c16ebce0badc26e3ad5d0ee#diff-3f790987dd77e1f06baa6dcaf001d026de91ede112d783a6eb2e37b73fbc1885R251)